srcdeps-maven-plugin/src/main/java/org/srcdeps/mvn/plugin/SrcdepsUpgradeMojo.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.xml.serialize.OutputFormat;
+import org.apache.xml.serialize.XMLSerializer;
+import org.srcdeps.mvn.Constants;
+import org.w3c.dom.Document;
+
+/**
+ * Adds and/or upgrades the srcdeps core extensions in {@code .mvn/extensions.xml} file. If the
+ * {@code .mvn/extensions.xml} does not exist, it creates it from scratch. If the {@code .mvn/extensions.xml} exists, it
+ * adds missing srcdeps core extensions and ensures they have the version specified by {@link #newVersion} parameter.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+@SuppressWarnings("deprecation")
+@Mojo(name = "up", defaultPhase = LifecyclePhase.NONE, threadSafe = false, requiresProject = true, requiresDependencyResolution = ResolutionScope.NONE)
+public class SrcdepsUpgradeMojo extends AbstractMojo {
+    protected final File multiModuleRootDir = new File(System.getProperty("maven.multiModuleProjectDirectory"));
+    protected final Path mvnDir;
+    protected final Path extensionsXmlPath;
+
+    /** The encoding to use when reading and writing {@code srcdeps.yaml} and {@code extensions.xml} files */
+    @Parameter(defaultValue = "utf-8", property = "srcdeps.encoding")
+    protected String encoding;
+
+    public SrcdepsUpgradeMojo() {
+        super();
+        this.mvnDir = multiModuleRootDir.toPath().resolve(".mvn");
+        this.extensionsXmlPath = mvnDir.resolve("extensions.xml");
+    }
+
+    /**
+     * The version of srcdeps core extensions to add or upgarde to in {@code extensions.xml} file. Default is the
+     * version if the {@code srcdeps-maven-plugin} that runs this mojo.
+     */
+    @Parameter(property = "srcdeps.newVersion")
+    protected String newVersion;
+
+    /** The limit for the word wrap in {@code extensions.xml} file */
+    @Parameter(defaultValue = "128", property = "srcdeps.xmlLineLength")
+    protected int xmlLineLength;
+
+    /** The number of spaces to use for one level of indentation in {@code extensions.xml} file */
+    @Parameter(defaultValue = "2", property = "srcdeps.xmlIndentSize")
+    protected int xmlIndentSize;
+
+    /**
+     * If {@code true} the {@code extensions.xml} file will be indented; otherwise the {@code extensions.xml} file will
+     * not be indented.
+     */
+    @Parameter(defaultValue = "true", property = "srcdeps.xmlIndent")
+    protected boolean xmlIndent;
+
+    /** If {@code true} the execution of this mojo will be skipped altogether; otherwise this mojo will be executed. */
+    @Parameter(defaultValue = "false", property = "srcdeps.skip")
+    protected boolean skip;
+
+    @Parameter(defaultValue = "${session}")
+    protected MavenSession session;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        final Log log = getLog();
+        if (skip || !multiModuleRootDir.equals(session.getCurrentProject().getBasedir())) {
+            log.info(getClass().getSimpleName() + " skipped");
```

Comment:
Personally, I find it cleaner to make a "return" and let the remaining of the code outside the if/else on such situation. It reduces the complexity when reading the code, as you don't have to remember "what's the if for this else again?"